frontend/resources/include/GUI_Resource.h

```diff
@@ -0,0 +1,20 @@
+/*
+ * GUI_Resource.h
+ *
+ * Copyright (C) 2020 by VISUS (Universitaet Stuttgart).
+ * Alle Rechte vorbehalten.
+ */
+
+#pragma once
+
+namespace megamol {
+namespace frontend_resources {
+
+struct GUIResource {
+
+    // Function requesting updated GUI state (e.g. ScreenshotService save project with GUI state tp PNG header)
+    std::function<std::string(void)> request_gui_state = nullptr;
```

Comment:
I think it is easier to have a functor that always returns something, i.e. an empty string could also be a valid GUI state. Then nobody needs to check for a null pointer.